### PR TITLE
Port to github actions and add pre-commit

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
- Port from travis to github actions
- Add pre-commit
- Initial fixes to files with pre-commit

### CODE REVIEW

**Comment by vatanaksoytezer** [comment]
I can remove travis when this is ready to be merged

**Comment by vatanaksoytezer** [comment]
I am not %100 sure about website deploy, since I couldn't test it

File: README.md [hunk 1]
```diff
@@ -16,11 +16,11 @@ This repository is currently built automatically by two systems. Travis builds t
 - [![ROS Melodic Build Farm Status](http://build.ros.org/buildStatus/icon?job=Mdoc__moveit_tutorials__ubuntu_bionic_amd64)](http://build.ros.org/job/Mdoc__moveit_tutorials__ubuntu_bionic_amd64/) [ROS Melodic Build Farm](http://docs.ros.org/melodic/api/moveit_tutorials/html/)
 - [![ROS Kinetic Build Farm Status](http://build.ros.org/buildStatus/icon?job=Kdoc__moveit_tutorials__ubuntu_xenial_amd64)](http://build.ros.org/job/Kdoc__moveit_tutorials__ubuntu_xenial_amd64/) [ROS Kinetic Build Farm](http://docs.ros.org/kinetic/api/moveit_tutorials/html/)
```
Comment (MarqRazz) [inline README.md:17]: Can we add the badge for each action here?

File: README.md [hunk 2]
```diff
@@ -16,11 +16,11 @@ This repository is currently built automatically by two systems. Travis builds t
...
 
-## Getting Started 
-An issue has been created for each tutorial to be ported to Foxy. At the top of each tutorial there is a tag: ":moveit1:", remove the tag 
+## Getting Started
+An issue has been created for each tutorial to be ported to Foxy. At the top of each tutorial there is a tag: ":moveit1:", remove the tag
 after the tutorial has been successfully updated.
 
-Below are some links to help with the ports.
```
Comment (MarqRazz) [inline README.md:23]: Should we add some documentation about installing and setting up pre-commit for developers?

**Review by MarqRazz — COMMENTED** [review]
Can the user run any other tests locally besides `format`?

Are you going to remove Travis in a different PR?

**Comment by vatanaksoytezer** [comment]
> Can the user run any other actions locally besides `format`?

They can actually run industrial_ci or any action with https://github.com/nektos/act in theory.

> Are you going to remove Travis in a different PR?

I will remove it in this PR, I just wanted to be sure that everything is OK before I add the commit that removes travis.

File: README.md (hunk 2 above)
Comment (vatanaksoytezer) [reply README.md:23]: I would %100 agree with that. But imo it is better that we add pre-commit stuff under "Contributing" section of README of this repo.

Comment (MarqRazz) [reply README.md:23]: Absolutely, sorry I was just in that section where there were changes and was being lazy :)

Comment (vatanaksoytezer) [reply README.md:23]: Yep, no worries on that. I will add it to this PR today.

**Comment by vatanaksoytezer** [comment]
This PR is now ready to be merged! @JafarAbdi @MarqRazz

**Review by MarqRazz — APPROVED** [review]
@vatanaksoytezer this is good to go, great job!

We should make sure the webpage deploys before we merge anything else. When your ready I can change the branch protection rules to remove Travis and add the new actions.

**Comment by vatanaksoytezer** [comment]
> We should make sure the webpage deploys before we merge anything else. When your ready I can change the branch protection rules to remove Travis and add the new actions.

Yep we are good to go! Tested in my gh-pages, seems to deploy without any problems.

**Pull request merged**